src/main/java/com/facebook/openwifirrm/ucentral/UCentralUtils.java

```diff
@@ -42,35 +44,55 @@ public class UCentralUtils {
 	private UCentralUtils() {}
 
 	/** Represents a single entry in wifi scan results. */
-	public static class WifiScanEntry {
-		public int channel;
-		public long last_seen;
-		public int signal;
-		public String bssid;
-		public String ssid;
-		public long tsf;
-		public String ht_oper;
-		public String vht_oper;
-		public int capability;
-		public int frequency;
-		public JsonArray ies;
+	public static class ProcessedWifiScanEntry extends WifiScanEntry {
 
-		/** Default Constructor. */
-		public WifiScanEntry() {}
+		/**
+		 * Unix time in milliseconds (ms). This field is not defined in the uCentral
+		 * API. We added it because it was unclear whether the {@code tsf} field was a
+		 * reliable time measurement (e.g., we did not know if/when it got reset or if
+		 * it is always available).
+		 */
+		public long unixTimeMs;
+
+		/** Default constructor. */
+		public ProcessedWifiScanEntry() {
+		}
 
 		/** Copy Constructor. */
-		public WifiScanEntry(WifiScanEntry o) {
-			this.channel = o.channel;
-			this.last_seen = o.last_seen;
-			this.signal = o.signal;
-			this.bssid = o.bssid;
-			this.ssid = o.ssid;
-			this.tsf = o.tsf;
-			this.ht_oper = o.ht_oper;
-			this.vht_oper = o.vht_oper;
-			this.capability = o.capability;
-			this.frequency = o.frequency;
-			this.ies = o.ies;
+		public ProcessedWifiScanEntry(ProcessedWifiScanEntry o) {
+			super(o);
+			this.unixTimeMs = o.unixTimeMs;
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (!super.equals(obj)) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+			ProcessedWifiScanEntry other = (ProcessedWifiScanEntry) obj;
+			return unixTimeMs == other.unixTimeMs;
+		}
+
+		// TODO should I NOT implement hashCode since this is mutable, or I SHOULD
+		// because implemented equals()?
+		@Override
+		public int hashCode() {
+			final int prime = 31;
+			int result = super.hashCode();
+			result = prime * result + Objects.hash(unixTimeMs);
+			return result;
+		}
+
+		@Override
+		public String toString() {
+			return String.format("ProcessedWifiScanEntry[signal=%d, bssid=%s, unixTimeMs=%d]", signal, bssid,
+					unixTimeMs);
```

Comment:
optional : format `unixTimeMs` into human-readable string (https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html)

Reply:
I think I will leave this as is for now since I use it more for debugging for now, where the raw value is easier to use :)